source/developers/development_environment.markdown

```diff
@@ -11,20 +11,55 @@ footer: true
 
 You'll need to set up a development environment if you want to develop a new feature or component for Home Assistant. Read on to learn how to set up.
 
-* Visit the [Home Assistant repository](https://github.com/home-assistant/home-assistant) and click **Fork**.
-
-* Consider setting up a virtual environment using [`venv`](https://docs.python.org/3.4/library/venv.html) before running the setup script.
+### {% linkable_title Setup Local Repository %}
 
+Visit the [Home Assistant repository](https://github.com/home-assistant/home-assistant) and click **Fork**.
+Once forked, setup your local copy of the source using the commands:
 ```bash
 $ git clone https://github.com/YOUR_GIT_USERNAME/home-assistant.git
 $ cd home-assistant
 $ git remote add upstream https://github.com/home-assistant/home-assistant.git
+```
+
+### {% linkable_title Prepare Your Environment %}
+
+#### {% linkable_title Core dependencies %} 
+
+Install the core dependencies.
+```bash
+$ sudo apt-get install python3-pip python3-dev libssl-dev
+```
+
+#### {% linkable_title Frontend dependencies (optional) %} 
+
+In order to run or develop the [Frontend](https://home-assistant.io/developers/frontend/) you will need node.js. The preferred method of installing node is [nvm](https://github.com/creationix/nvm). Install nvm using the instructions in the [README](https://github.com/creationix/nvm#install-script), and install node.js by running the following command:
+```bash
+$ nvm install node
+```
+
+#### {% linkable_title Setting up virtual environment (optional) %} 
+
+If you plan on providing isolation to your environment using [`venv`](https://docs.python.org/3.4/library/venv.html). You'll need to install the python3-venv package:
+```bash
+$ sudo apt-get install python3-venv
+```
+Within the `home-assistant` directory, create and activate your virtual environment using the commands:
+```bash
+$ pyvenv venv
```

Comment:
By specifying venv, it places all the virtual environment in a separate directory. One which is configured in the .gitignore file to be excluded. Otherwise your home-assistant directory is populated/polluted by:
```bash
martinr@ha-dev:~/home-assistant-backup/venv$ ls -go
total 28
drwxr-xr-x 3 4096 Jan 18 11:50 bin
drwxr-xr-x 2 4096 Jan 18 11:50 include
drwxr-xr-x 5 4096 Jan 18 11:50 lib
lrwxrwxrwx 1   37 Jan 18 11:50 lib64 -> /home/martinr/home-assistant/venv/lib
-rw-r--r-- 1   60 Jan 18 11:50 pip-selfcheck.json
-rw-r--r-- 1   69 Jan 18 11:50 pyvenv.cfg
drwxr-xr-x 3 4096 Jan 18 11:50 selenium
drwxr-xr-x 4 4096 Jan 18 11:50 share
```
Does that make sense?